runelite-client/src/main/java/net/runelite/client/plugins/bosslogger/BossLoggerPlugin.java

```diff
@@ -0,0 +1,1365 @@
+
+	@Inject
+	private Notifier notifier;
+
+	@Inject
+	private ChatMessageManager chatMessageManager;
+
+	@Inject
+	private PluginToolbar pluginToolbar;
+
+	private File playerFolder;	// Where we are storing files
+	private static final Pattern NUMBER_PATTERN = Pattern.compile("([0-9]+)");
+	private static final Pattern BOSS_NAME_PATTERN = Pattern.compile("Your (.*) kill count is:");
+	private static final Pattern PET_RECEIVED_PATTERN = Pattern.compile("You have a funny feeling like ");
+	private static final Pattern PET_RECEIVED_INVENTORY_PATTERN = Pattern.compile("You feel something weird sneaking into your backpack.");
+	private String messageColor = ""; // in-game chat message color
+
+	private BossLoggerPanel panel;
+	private NavigationButton navButton;
+
+	// Mapping Variables
+	private Map<String, Boolean> recordingMap = new HashMap<>(); 			// Store config recording value by tab boss name
+	private Map<String, ArrayList<LootEntry>> lootMap = new HashMap<>();	// Store loot entries by boss name
+	private Map<String, Integer> killcountMap = new HashMap<>(); 			// Store boss kill count by name
+	private Map<String, String> filenameMap = new HashMap<>(); 				// Stores filename for each boss name
+
+	// Loot Finding Variables
+	private Set<Tile> changedItemLayerTiles = new HashSet<Tile>();				// Stores tiles that have had ItemLayer changes
+	private WorldPoint[] deathLocations;										// Stores NPC Death Worldpoints
+	private Map<WorldPoint, Map<Integer, Integer>> itemArray = new HashMap<>();	// Stores item map for a specific WorldPoint
+	private Map<Actor, Actor> actors = new HashMap<>();							// Stores Actors we've interacted with to check for their deaths
+
+	// Variables for handling NPC death
+	private String deathName;				// NPC Name
+	private String lastBossKilled;			// NPC Name, saved for potential pet logging issues
+	private int deathSize;					// NPC Size
+	private int deathID;					// NPC ID
+	private WorldPoint deathSpot;			// NPC Death WorldPoint
+	private WorldPoint playerLocation;		// Players WorldPoint at time of NPC death animation
+	private boolean gotPet = false;			// Got the pet chat message?
+
+	// Watching flags (actor death/changed item layer)
+	private boolean watching = false;				// Watching for ActorDespawn?
+	private boolean watchingItemLayers = false;		// Watching for ItemLayerChanged?
+
+	@Provides
+	BossLoggerConfig provideConfig(ConfigManager configManager)
+	{
+		return configManager.getConfig(BossLoggerConfig.class);
+	}
+
+	@Override
+	protected void startUp()
+	{
+		if (bossLoggerConfig.showLootTotals())
+		{
+			// Waits 2 seconds, helps ensure itemManager is loaded
+			// Client cache loading is async, plugins can be loaded before it is finished
+			ScheduledExecutorService scheduler = Executors.newScheduledThreadPool(1);
+			scheduler.schedule(() -> SwingUtilities.invokeLater(this::createPanel), 2, TimeUnit.SECONDS);
+		}
+		init();
+
+		// Ensure Loot Directory has been created
+		LOOTS_DIR.mkdir();
```

Comment:
Worth verifying the return of this in order to handle a scenario where creating the directory failed